lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -375,6 +389,60 @@ function createPsuedoLocaleStrings(messages) {
   return psuedoLocalizedStrings;
 }
 
+/**
+ * @param {import('typescript').NamedDeclaration} node
+ * @return {string}
+ */
+function getIdentifier(node) {
+  if (!node.name || !tsc.isIdentifier(node.name)) throw new Error('no Identifier found');
+
+  return node.name.text;
+}
+
+/**
+ * @param {string} sourceStr String of the form 'const UIStrings = {...}'.
+ * @param {Record<string, string>} liveUIStrings The actual imported UIStrings object.
+ * @return {Record<string, ParsedUIString>}
+ */
+function parseUIStrings(sourceStr, liveUIStrings) {
+  const tsAst = tsc.createSourceFile('uistrings', sourceStr, tsc.ScriptTarget.ES2019, true, tsc.ScriptKind.JS);
+
+  const extractionError = new Error('UIStrings was not extracted correctly by the collect-strings regex.');
+  const uiStringsStatement = tsAst.statements[0];
+  if (tsAst.statements.length !== 1) throw extractionError;
+  if (!tsc.isVariableStatement(uiStringsStatement)) throw extractionError;
+
+  const uiStringsDeclaration = uiStringsStatement.declarationList.declarations[0];
+  if (!tsc.isVariableDeclaration(uiStringsDeclaration)) throw extractionError;
+  if (getIdentifier(uiStringsDeclaration) !== 'UIStrings') throw extractionError;
+
+  const uiStringsObject = uiStringsDeclaration.initializer;
+  if (!uiStringsObject || !tsc.isObjectLiteralExpression(uiStringsObject)) throw extractionError;
+
+  /** @type {Record<string, ParsedUIString>} */
+  const parsedMessages = {};
+
+  for (const property of uiStringsObject.properties) {
+    const key = getIdentifier(property);
+
+    // Use live message to avoid having to e.g. concat strings broken into parts.
```

Comment:
I suspect this is why @patrickhulce did this in the first place, but I'm not sure :)

Reply:
yes, precisely :) I think I got like 20% of the way done until I hit the first file that split a line and was like f' it let's just `require` 😆